### PR TITLE
registry-console: set node selector to registry-console deployment

### LINKED ISSUE #8438: the cockpit-ui role deploys to compute nodes and not infra nodes
#### Description

Registry Console deployed via `openshift-ansible`'s `roles/cockpit-ui` role which uses the `registry-console` template as deployed via the `roles/openshift_hosted_templates` role and that template does not have a `nodeSelector` field in it as of the `openshift-ansible` `3.9.29` release. I'm unsure of how to resolve this in the best way because I'm unfamiliar with the implications of adding that field to the generic template that is destined for the Service Catalog.

##### Version

```
ansible 2.4.3.0                            
  config file = /usr/share/atomic-openshift-utils/ansible.cfg                         
  configured module search path = [u'/opt/app-root/src/.ansible/plugins/modules', u'/usr/share/ansible/plugins/modules']                                                     
  ansible python module location = /usr/lib/python2.7/site-packages/ansible           
  executable location = /usr/bin/ansible   
  python version = 2.7.5 (default, Feb 20 2018, 09:19:12) [GCC 4.8.5 20150623 (Red Hat 4.8.5-28)]
```

```
openshift-ansible-3.9.14-1.git.3.c62bc34.el7.noarch
```

##### Expected Results
The Registry console pods to run on infra nodes.

##### Observed Results
Registry console runs on compute nodes.

##### Additional Information

I'm not sure what the best course of action here is because I don't know all the implications of either modifying the `registry-console` template in `roles/openshift_hosted_templates` or changing the `nodeSelector` elsewhere but I would like a way to define what the `nodeSelector` will be at the point in time the Registry Console is deployed.

### PR DESCRIPTION
**Pull request by vrutkovs**
Allow limiting registry-console to particular nodes, infra by default

Fixes #8438

### CODE REVIEW

File: roles/cockpit-ui/defaults/main.yml [hunk 1]
```diff
@@ -1,2 +1,3 @@
 ---
 openshift_config_base: "/etc/origin"
+openshift_hosted_registry_console_nodeselector: "{{ openshift_hosted_infra_selector | default('{\"node-role.kubernetes.io/infra\": "true"}') }}"
```
[the remark this answers is not in the document]
Comment (vrutkovs) [reply roles/cockpit-ui/defaults/main.yml]: Reworked this to use `map_from_pairs`

File: roles/openshift_hosted_templates/files/v3.10/enterprise/registry-console.yaml [hunk 2]
```diff
@@ -25,6 +25,7 @@ objects:
           labels:
             name: "registry-console"
         spec:
+          nodeSelector: ${{NODE_SELECTOR}}
```
Comment (michaelgugino) [inline roles/openshift_hosted_templates/files/v3.10/enterprise/registry-console.yaml]: Should be single curl braces.

File: roles/openshift_hosted_templates/files/v3.10/origin/registry-console.yaml [hunk 3]
```diff
@@ -25,6 +25,7 @@ objects:
           labels:
             name: "registry-console"
         spec:
+          nodeSelector: "${{NODE_SELECTOR}}"
```
Comment (michaelgugino) [inline roles/openshift_hosted_templates/files/v3.10/origin/registry-console.yaml]: Single curly braces.

File: roles/cockpit-ui/tasks/main.yml [hunk 4]
```diff
@@ -23,41 +23,38 @@
...
-    command: mktemp -d /tmp/openshift-ansible-XXXXXX
-    register: mktemp
-    changed_when: False
+  - name: Set registry-console params
+    set_fact:
+      l_registry_console_params:
+        OPENSHIFT_OAUTH_PROVIDER_URL: "{{ openshift.master.public_api_url }}"
+        REGISTRY_HOST: "{{ docker_registry_route.results[0].spec.host }}"
+        COCKPIT_KUBE_URL: "https://{{ registry_console_cockpit_kube.results.results[0].spec.host }}"
+        NODE_SELECTOR: "{{ openshift_hosted_registry_console_nodeselector | map_from_pairs }}"
```
Comment (michaelgugino) [inline roles/cockpit-ui/tasks/main.yml]: Default value for this variable appears to be a dictionary.  Does map_from_pairs handle this case?

File: roles/openshift_hosted_templates/files/v3.10/enterprise/registry-console.yaml (hunk 2 above)
Comment (vrutkovs) [reply roles/openshift_hosted_templates/files/v3.10/enterprise/registry-console.yaml]: This is important, it means it a json/yaml, not string - see https://docs.openshift.org/latest/dev_guide/templates.html#writing-parameters

File: roles/cockpit-ui/tasks/main.yml (hunk 4 above)
Comment (vrutkovs) [reply roles/cockpit-ui/tasks/main.yml]: Defaults for both `openshift_hosted_registry_console_nodeselector` and `openshift_hosted_infra_selector` are strings.

Updated `hosts.example`

File: roles/cockpit-ui/tasks/main.yml [hunk 5]
```diff
@@ -23,41 +23,38 @@
...
-    - "'already exists' not in deploy_registry_console.stderr"
-    - "deploy_registry_console.rc != 0"
-
-  - name: Delete temp directory
-    file:
-      name: "{{ mktemp.stdout }}"
-      state: absent
-    changed_when: False
-    # XXX: End required for items still using command
+    oc_process:
```
Comment (michaelgugino) [inline roles/cockpit-ui/tasks/main.yml]: Based on Clayton's feedback in the other PR, we probably shouldn't do this here either.

Comment (vrutkovs) [reply roles/cockpit-ui/tasks/main.yml]: This PR is fine - the template is already there, provided by `openshift_hosted_templates` role.

I've set WIP flag on other PRs, which created templates

**Comment by kwoodson** [comment]
@vrutkovs, What's the status of this PR?  I'm still seeing the registry-console land on non-infra nodes.  This PR is required as there is currently no option to land a `nodeSelector` on the registry-console through the installer.

**Comment by vrutkovs** [comment]
> What's the status of this PR?

This needs some work around single/double curly braces (I think double are required, but didn't test it yet) and another review

**Comment by vrutkovs** [comment]
Replaced by #8766 - the node selector should be masters and not be configurable

**Pull request closed without merging**